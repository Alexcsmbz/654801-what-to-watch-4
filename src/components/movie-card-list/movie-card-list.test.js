import MovieCardList from './movie-card-list.tsx';
import {BrowserRouter as Router} from 'react-router-dom';

const movies = [
  {
    name: `movie 1`,
    img: `img/img-1.jpg`,
  },
  {
    name: `movie 2`,
    img: `img/img-2.jpg`,
  },
  {
    name: `movie 3`,
    img: `img/img-3.jpg`,
  },
  {
    name: `movie 4`,
    img: `img/img-4.jpg`,
  },
];
const movieShow = 8;

describe(`MovieCardList snapshot test`, () => {
  it(`MovieCardList should render movie list`, () => {
    const tree = renderer.create(
        <Router>
          <MovieCardList
            movies={movies}
            movieShow={movieShow}
          />
        </Router>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
