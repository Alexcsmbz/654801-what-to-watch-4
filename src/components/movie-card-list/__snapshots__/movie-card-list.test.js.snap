// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieCardList snapshot test MovieCardList should render movie list 1`] = `
Array [
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
  >
    <div
      onClick={[Function]}
    >
      <video
        controls={true}
        height="175"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        width="280"
      >
        <source
          type="video/mp4"
        />
      </video>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/undefined"
          onClick={[Function]}
        >
          movie 1
        </a>
      </h3>
    </div>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
  >
    <div
      onClick={[Function]}
    >
      <video
        controls={true}
        height="175"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        width="280"
      >
        <source
          type="video/mp4"
        />
      </video>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/undefined"
          onClick={[Function]}
        >
          movie 2
        </a>
      </h3>
    </div>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
  >
    <div
      onClick={[Function]}
    >
      <video
        controls={true}
        height="175"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        width="280"
      >
        <source
          type="video/mp4"
        />
      </video>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/undefined"
          onClick={[Function]}
        >
          movie 3
        </a>
      </h3>
    </div>
  </article>,
  <article
    className="small-movie-card catalog__movies-card"
    onClick={[Function]}
  >
    <div
      onClick={[Function]}
    >
      <video
        controls={true}
        height="175"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        width="280"
      >
        <source
          type="video/mp4"
        />
      </video>
      <h3
        className="small-movie-card__title"
      >
        <a
          className="small-movie-card__link"
          href="/films/undefined"
          onClick={[Function]}
        >
          movie 4
        </a>
      </h3>
    </div>
  </article>,
]
`;
